src/ecs/resources/input/button.rs

```diff
@@ -0,0 +1,29 @@
+use engine::{VirtualKeyCode, MouseButton};
+
+use super::local_mouse_button::LocalMouseButton;
+use super::local_virtual_key_code::LocalVirtualKeyCode;
+
+/// A Button is any kind of digital input that the engine supports.
+#[derive(Eq, PartialEq, Debug, Copy, Clone, Serialize, Deserialize)]
+pub enum Button {
+    /// Keyboard keys
+    Key(#[serde(with = "LocalVirtualKeyCode")]
```

Comment:
Given that we already got our own input structs, would it make sense to just get rid of glutin types in the public API and use our own types?